src/Service.Host/client/src/components/PartStorageType.js

```diff
@@ -0,0 +1,268 @@
+import React, { useEffect, useState } from 'react';
+import Typography from '@mui/material/Typography';
+import { useParams } from 'react-router';
+import Grid from '@mui/material/Grid2';
+import PropTypes from 'prop-types';
+import {
+    Loading,
+    Search,
+    InputField,
+    ErrorCard,
+    SnackbarMessage
+} from '@linn-it/linn-form-components-library';
+import Button from '@mui/material/Button';
+import Page from './Page';
+import config from '../config';
+import itemTypes from '../itemTypes';
+import useInitialise from '../hooks/useInitialise';
+import useSearch from '../hooks/useSearch';
+import usePut from '../hooks/usePut';
+import usePost from '../hooks/usePost';
+
+function PartStorageType({ creating }) {
+    const { id } = useParams();
+
+    const { isPartStorageTypesLoading, result: partStorageTypeResult } = useInitialise(
+        `${itemTypes.partsStorageTypes.url}/${id}`
+    );
+
+    const [partStorageType, setPartStorageType] = useState(partStorageTypeResult);
+
+    const [partSearchTerm, setPartSearchTerm] = useState('');
+    const [storageTypeSearchTerm, setStorageTypeSearchTerm] = useState('');
+    const [snackbarVisible, setSnackbarVisible] = useState();
+
+    useEffect(() => {
+        setPartStorageType(partStorageTypeResult);
+    }, [partStorageTypeResult]);
+
+    const {
+        search: searchParts,
+        results: partsSearchResults,
+        loading: partsSearchLoading,
+        clear: clearParts
+    } = useSearch(itemTypes.parts.url, 'partNumber', 'partNumber', 'description');
+
+    const {
+        search: searchStorageTypes,
+        results: storageTypesSearchResults,
+        loading: storageTypesSearchLoading,
+        clear: clearStorageTypes
+    } = useSearch(itemTypes.storageTypes.url, 'storageTypeCode', 'storageTypeCode', 'description');
+
+    const {
+        send: updatePartStorageType,
+        isLoading: updateLoading,
+        errorMessage: updateError,
+        putResult: updateResult,
```

Comment:
generally a good idea to use the new value of updateResult to update the page with the 'updated' data (even if it is the same as the values the user just entered that are currently on screen - Imagine though for example your backend code did some tidying, toUpperCase()'ing etc of strings - Ideally you want the user to see exactly the updated state of the database row on their screen when they get a 'save succesful' message, not just whatever state they happened to leave it in)